src/fsearch_preferences_ui.c

```diff
@@ -163,23 +163,17 @@ on_file_chooser_native_dialog_response(GtkNativeDialog *dialog, GtkResponseType
 
     if (response == GTK_RESPONSE_ACCEPT) {
         GtkFileChooser *chooser = GTK_FILE_CHOOSER(dialog);
-        GSList *filenames = gtk_file_chooser_get_filenames(chooser);
-        if (filenames) {
-            for (GSList *f = filenames; f != NULL; f = f->next) {
-                gchar *filename = f->data;
-                if (filename) {
-                    ctx->add_path_cb(ctx->model, filename);
-                }
+        GListModel *filenames = gtk_file_chooser_get_files(chooser);
```

Comment:
We can use the `g_autoptr` macro here to automatically free `filenames` when it gets out of scope — then the corresponding `g_object_unref` can be removed:
`g_autoptr(GListModel) filenames = ...`